preprocessing/shift_generation.py

```diff
@@ -310,11 +351,11 @@ def already_daily_off_shift(root, employee_offset, employee_rest, day):
             temp_val = interval[1]
             interval[1] = temp_val + employee_offset
         if (
-            employee_offset + (24 * int(day))
+            24 * int(day)
             <= interval[0]
             <= employee_offset + (24 * (int(day) + 1))
         ):
-            if interval[1] - interval[0] >= employee_rest:
+            if interval[1] - interval[0] - max(employee_offset - interval[0], 0) >= employee_rest:
```

Comment:
What is the intention here?

Reply:
If there is a no_demand_interval from 00:00 - 14:00, and you have an employee_offset of 4 hours: 

The previous code would not consider this interval, because 00:00 is NOT >= employee_offset. The intention is to see whether there exist daily_rest naturally, and for this instance it does (employee specific day starts 04:00, first demand occcurs 14:00 --> 10 hours naturally rest that employee specific day).

In the updated code we include demand intervals that starts before employee_offset, and instead take the employee_offset into account later. By doing this, the max-term must be introduced. In the example above, employee_offset - inteval[0] = 4. This leads to the value interval[1] - [inteval[0] - max(4, 0) = 14 - 0 - 4 = 10 = correct.

But what if there only was a no_demand from 06:00 - 14:00. Then 06:00 >= 00:00, but employee_offset - inteval[0] = 04:00 - 06:00 = - 2. This would lead to interval[1] - [inteval[0] - (-2) = 14 - 6 + 2 = 10 = incorrect. Therefore the max is introduced, turning -2 into 0 so that the calculation becoms correct.